### PR TITLE
Clair notification handler

### PR DESCRIPTION
**Pull request by reasonerjt**

### CODE REVIEW

File: make/common/templates/clair/config.yaml [hunk 1]
```diff
@@ -16,8 +16,10 @@ clair:
     # Deadline before an API request will respond with a 503
     timeout: 300s
   updater:
-    interval: 2h
+    interval: 1h
```
Comment (yixingjia) [inline make/common/templates/clair/config.yaml]: We properly need to change this number to a larger one.

Comment (reasonerjt) [inline make/common/templates/clair/config.yaml]: Thanks, this is for testing, so Clair will generate notifications more frequently, I'll open another issue to put interval to 12h in next sprint.

File: src/ui/service/notifications/clair/handler.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+// returns true to indicate it should reshedule the "rescan" action.
+func (t *timer) needReschedule() bool {
+	t.Lock()
+	defer t.Unlock()
+	if time.Now().Before(t.next) {
+		return false
+	}
+	t.next = time.Now().Add(rescanInterval)
```
Comment (ywk253100) [inline src/ui/service/notifications/clair/handler.go]: I don't know what the frequency that the notifications will be received is. If it's high, maybe we can put the lock operation after the if statement.

File: src/ui/service/notifications/clair/handler.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	var ne models.ClairNotificationEnvelope
+	if err := json.Unmarshal(h.Ctx.Input.CopyBody(1<<32), &ne); err != nil {
+		log.Errorf("Failed to decode the request: %v", err)
+		return
+	}
+	log.Debugf("Received notification from Clair, name: %s", ne.Notification.Name)
+	notification, err := clairClient.GetNotification(ne.Notification.Name)
+	if err != nil {
+		log.Errorf("Failed to get notification details from Clair, name: %s, err: %v", ne.Notification.Name, err)
+	}
```
Comment (ywk253100) [inline src/ui/service/notifications/clair/handler.go]: return?

File: src/ui/service/notifications/clair/handler.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	if new := notification.New; new != nil {
+		if vuln := new.Vulnerability; vuln != nil {
+			log.Debugf("new vulnerability namespace: %s", vuln.NamespaceName)
+			ns[vuln.NamespaceName] = true
+		}
+	}
+	for k, v := range ns {
+		if v {
+			if err := dao.SetClairVulnTimestamp(k, time.Now()); err != nil {
+				log.Debugf("Updated the timestamp for namespaces: %s", k)
```
Comment (ywk253100) [inline src/ui/service/notifications/clair/handler.go]: err == nil

File: src/ui/service/notifications/clair/handler.go (hunk 3 above)
Comment (reasonerjt) [inline src/ui/service/notifications/clair/handler.go]: right...

File: src/ui/service/notifications/clair/handler.go (hunk 4 above)
Comment (reasonerjt) [inline src/ui/service/notifications/clair/handler.go]: right...

File: src/ui/service/notifications/clair/handler.go (hunk 2 above)
Comment (reasonerjt) [inline src/ui/service/notifications/clair/handler.go]: But Add may not be atomic then Before may accepts an incomplete data as parm?

**Pull request merged**